website/docs/d/data_lake_pipeline_run.html.markdown

```diff
@@ -0,0 +1,49 @@
+---
+layout: "mongodbatlas"
+page_title: "MongoDB Atlas: data_lake_pipeline_run"
+sidebar_current: "docs-mongodbatlas-datasource-data-lake-pipeline-run"
+description: |-
+    Describes a Data Lake Pipeline Run.
+---
+
+# Data Source: mongodbatlas_data_lake_pipeline_run
+
+`mongodbatlas_data_lake_pipeline_run` describe a Data Lake Pipeline Run.
+
+
+-> **NOTE:** Groups and projects are synonymous terms. You may find `groupId` in the official documentation.
+
+## Example Usage
+
+```terraform
+data "mongodbatlas_data_lake_pipeline_run" "test" {
+  project_id = "PROJECT ID"
+  name = "DATA LAKE PIPELINE NAME"
+  pipeline_run_id = "DATA LAKE PIPELINE RUN ID"
```

Comment:
Could we help the user out here a bit more?   For example, I think some will struggle to put together that they will first need to use data_lake_pipeline_runs for the name (which they will get from when they setup mongodbatlas_data_lake_pipeline) and that will give them the run id, e.g.:

```
resource "mongodbatlas_data_lake_pipeline" "pipeline" {
      //assuming we've already setup project and cluster in another block
      project_id       = mongodbatlas_project.projectTest.project_id
      name       = "DataLakePipelineName"
      sink {
        type = "DLS"
        partition_fields {
            name = "access"
            order = 0
        }
      }
      source {
        type = "ON_DEMAND_CPS"
        cluster_name = mongodbatlas_cluster.clusterTest.name
        database_name = "sample_airbnb"
        collection_name = "listingsAndReviews"
      }
      transformations {
              field = "test"
              type  = "EXCLUDE"
      }
      transformations {
              field = "test22"
              type  = "EXCLUDE"
      }
}

data "mongodbatlas_data_lake_pipeline_runs" "pipeline_run" {
      project_id       = mongodbatlas_project.projectTest.project_id
     name = mongodbatlas_data_lake_pipeline.pipeline.name
}

data "mongodbatlas_data_lake_pipeline_run" "test" {
   project_id       = mongodbatlas_project.projectTest.project_id
  name = mongodbatlas_data_lake_pipeline.pipeline.name
  pipeline_run_id = mongodbatlas_data_lake_pipeline_runs.pipeline_run.pipeline_run_id
}

```

Reply:
I will add `mongodbatlas_data_lake_pipeline`  to the example. However, based on my local testing, `mongodbatlas_data_lake_pipeline` won't return a `run_id` but the user has to go in the UI and manually run the pipeline

Reply:
Updated

Reply:
@andreaangiolillo you bring up a good point about how this works.  Since runs aren't something that really  makes sense, from our scope review, to be managed by Terraform we've included them as a data source but not as a resource.  But one can trigger an "On Demand Pipeline Run" via the UI, as you note, and [API](https://www.mongodb.com/docs/atlas/reference/api-resources-spec/#tag/Data-Lake-Pipelines/operation/triggerSnapshotIngestion) but unless a scheduled run or ondemand run are going you won't get a run_id back.  So what I would suggest is to add "mongodbatlas_data_lake_pipeline_runs" to get the run_id but note in a comment something like `pipeline_run_id will only be returned if a schedule or ondemand run is active`  FYI @Zuhairahmed this could be an area where our users will guide us if what we have offered is not sufficient.

Reply:
Does `mongodbatlas_data_lake_pipeline_runs` make sense to include, if the user has to go to the UI and do the run to generate the id. At the point they're in the UI I expect it's just as easy for them to get the id and then provide that as a parameter for `mongodbatlas_data_lake_pipeline_run`

In particular, I find the naming here really confusing - `runs` vs `run`. It feels like we're forcing Terrafor mto handle an interactive flow that is better in an interactive tool like the cli